contrib/pkg/provision_aws/command.go

```diff
@@ -0,0 +1,233 @@
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package provision_aws
+
+import (
+	"fmt"
+	"os"
+	"os/user"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+
+	"k8s.io/apiserver/pkg/storage/names"
+	"k8s.io/client-go/discovery"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+
+	capiclient "sigs.k8s.io/cluster-api/pkg/client/clientset_generated/clientset"
+
+	clustopclient "github.com/openshift/cluster-operator/pkg/client/clientset_generated/clientset"
+)
+
+type ProvisionClusterOptions struct {
+	Namespace           string
+	Name                string
+	ImageFormat         string
+	RPMRepositoryURL    string
+	RPMRepositorySource string
+	SSHSecretName       string
+	AccountSecretName   string
+	KeyPairName         string
+	AnsibleImage        string
+	Region              string
+	KubeconfigFile      string
+	Version             string
+	VPCName             string
+	AMIVPCName          string
+	LogLevel            string
+	EtcdImage           string
+
+	ClusterName string
+	ImageID     string
+	Logger      log.FieldLogger
+}
+
+func NewProvisionClusterCommand() *cobra.Command {
+	userName := "user"
+	u, err := user.Current()
+	if err == nil {
+		userName = u.Username
+	}
+
+	opt := &ProvisionClusterOptions{}
+	cmd := &cobra.Command{
+		Use:   "provision-aws NAME",
+		Short: "Provision a new cluster with the given RPM repository and Image Format on AWS",
+		Run: func(cmd *cobra.Command, args []string) {
+			if err := opt.Complete(args); err != nil {
+				log.WithError(err).Error("Cannot complete command")
+				return
+			}
+			if err := opt.Validate(); err != nil {
+				log.WithError(err).Error("Invalid command options")
+				return
+			}
+			if err := opt.Run(); err != nil {
+				log.WithError(err).Error("Runtime error")
+			}
+		},
+	}
+	flags := cmd.Flags()
+	flags.StringVarP(&opt.Namespace, "namespace", "", "", "namespace to use. Defaults to current namespace.")
+	flags.StringVar(&opt.AccountSecretName, "account-secret", fmt.Sprintf("%s-aws-creds", userName), "secret to use for AWS account")
+	flags.StringVar(&opt.SSHSecretName, "ssh-secret", "libra-ssh", "secret with SSH key")
+	flags.StringVar(&opt.AnsibleImage, "ansible-image", "registry.svc.ci.openshift.org/openshift-cluster-operator/cluster-operator-ansible:latest", "ansible image to use for job")
+	flags.StringVar(&opt.RPMRepositoryURL, "rpm-repository", "", "RPM repository for Origin")
+	flags.StringVar(&opt.RPMRepositorySource, "rpm-repository-source", "https://gcsweb-ci.svc.ci.openshift.org/gcs/origin-ci-test/releases/openshift/origin/master/origin.repo", "pointer to RPM repository URL")
```

Comment:
I probably should know this but how do these two params differ?

Reply:
I think I see what's up from going through earlier code again, these could use some more detail in the description to clarify how they're used.